libsplinter/src/rest_api/mod.rs

```diff
@@ -789,6 +838,34 @@ impl RestApiBuilder {
         #[cfg(feature = "auth")]
         let identity_providers = self.identity_providers.unwrap_or_default();
 
+        #[cfg(feature = "oauth")]
+        let mut oauth_configuration = None;
+
+        #[cfg(feature = "oauth")]
+        match (self.oauth_client, self.save_tokens_operation) {
+            (Some(oauth_client), Some(save_tokens_operation)) => {
+                oauth_configuration = Some(OAuthConfiguration {
+                    oauth_client,
+                    save_tokens_operation,
+                });
+            }
+            (Some(_), None) => {
+                return Err(RestApiServerError::MissingField(
+                    "REST API auth is enabled as OAuth, but no save token operation has been \
+                    provided"
+                        .to_string(),
+                ));
+            }
+            (None, Some(_)) => {
+                return Err(RestApiServerError::MissingField(
+                    "REST API auth is not enabled as OAuth, but a save token operation has \
+                    been provided"
+                        .to_string(),
+                ));
+            }
+            (None, None) => (),
+        }
+
```

Comment:
This method is for starting the REST API without authentication, so this shouldn't be added.

Reply:
Fixed.